src/coreclr/vm/peassembly.cpp

```diff
@@ -296,10 +296,10 @@ void PEAssembly::OpenImporter()
     CONTRACTL_END;
 
     // Make sure internal MD is in RW format.
-    ConvertMDInternalToReadWrite();
+    ConvertMDInternalToReadWrite(openForWriting);
 
     IMetaDataImport2 *pIMDImport = NULL;
-    IfFailThrow(GetMetaDataPublicInterfaceFromInternal((void*)GetMDImport(),
+    IfFailThrow(GetMetaDataPublicInterfaceFromInternal((void*)m_pConvertedMDImport,
```

Comment:
Let's assert `m_pConvertedMDImport` is non NULL and comment why we need the converted version.